File: src/petrovich/util/rules/RulesLoader.java
/*$Id:$*/
package petrovich.util.rules;

import org.ho.yaml.Yaml;
import petrovich.util.rules.data.Rules;

import java.io.File;
import java.io.FileNotFoundException;

/**
 * @author DMITRY KNYAZEV
 * @since 31.05.2014
 */
public abstract class RulesLoader implements IRulesLoader
{
	public static Rules load() throws FileNotFoundException
	{
		return load(null);
	}

	public static Rules load( String path ) throws FileNotFoundException
	{
		File rule;
		if( path == null || path.isEmpty() )
			rule = new File("rules.yml");
		else
			rule = new File(path);
		//Map<String, Object> o = (Map<String, Object>) Yaml.load(rule);
		return Yaml.loadType(rule, Rules.class);
	}
}

File: src/main/kotlin/kt/petrovich/util/rules/RulesLoader.kt
package kt.petrovich.util.rules

import kt.petrovich.util.rules.data.Rules
import org.ho.yaml.Yaml

import java.io.File
import java.io.FileNotFoundException

/**
 * @author DMITRY KNYAZEV
 * @since 31.05.2014
 */
class RulesLoader private constructor() {
    init {
        throw AssertionError("It helper class. Create instance forbidden")
    }

    companion object {
        private val defaultRulesPath = Thread.currentThread()
                .contextClassLoader
                .getResource("rules.yml").path

        /**
         * Load rules from YAML file path
         *
         * @param path path to rules file
         * @return rules
         * @throws FileNotFoundException if file path not exist
         */
        @Throws(FileNotFoundException::class)
        @JvmOverloads
        fun loadRules(path: String = defaultRulesPath): Rules {
            return Yaml.loadType(File(path), Rules::class.java)
        }
    }
}
